controllers/machinedeployment_controller.go

```diff
@@ -220,15 +225,21 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, cluster *cl
 		return ctrl.Result{}, errors.Errorf("missing MachineDeployment strategy")
 	}
 
+	var requeueAfter time.Duration
+	if d.Spec.RolloutAfter != nil && d.Spec.RolloutAfter.After(r.reconciliationTime.Time) {
+		requeueAfter = d.Spec.RolloutAfter.Time.Sub(r.reconciliationTime.Time)
+		log.V(3).Info("RolloutAfter is scheduled", "rolloutafter", requeueAfter.String())
+	}
+
 	if d.Spec.Strategy.Type == clusterv1.RollingUpdateMachineDeploymentStrategyType {
 		if d.Spec.Strategy.RollingUpdate == nil {
 			return ctrl.Result{}, errors.Errorf("missing MachineDeployment settings for strategy type: %s", d.Spec.Strategy.Type)
 		}
-		return ctrl.Result{}, r.rolloutRolling(ctx, d, msList)
+		return ctrl.Result{RequeueAfter: requeueAfter}, r.rolloutRolling(ctx, d, msList)
 	}
 
 	if d.Spec.Strategy.Type == clusterv1.OnDeleteMachineDeploymentStrategyType {
-		return ctrl.Result{}, r.rolloutOnDelete(ctx, d, msList)
+		return ctrl.Result{RequeueAfter: requeueAfter}, r.rolloutOnDelete(ctx, d, msList)
```

Comment:
What's the reason to requeue?

Reply:
So when the rolloutAfter time is met it triggers reconciliation and the rollout takes place. Otherwise will happen at worst when the resync time happens which we might take as acceptable.

Reply:
Dropped now from the reconciler to avoid concurrency issues and dropped this